src/main/scala/dpla/ingestion3/mappers/providers/NaraMapping.scala

```diff
@@ -85,36 +86,40 @@ class NaraMapping extends XmlMapping with XmlExtractor {
-        case (Some("Restricted - Possibly"), Some("Donor Restrictions")) => Some(URI("http://rightsstatements.org/vocab/CNE/1.0/"))
-        case (Some("Restricted - Possibly"), Some("Public Law 101-246")) => Some(URI("http://rightsstatements.org/vocab/NKC/1.0/"))
-        case (Some("Restricted - Possibly"), Some("Service Mark")) => Some(URI("http://rightsstatements.org/vocab/NoC-OKLR/1.0/"))
-        case (Some("Restricted - Possibly"), Some("Trademark")) => Some(URI("http://rightsstatements.org/vocab/NoC-OKLR/1.0/"))
-        case (Some("Restricted - Possibly"), Some("Other")) => Some(URI("http://rightsstatements.org/vocab/NoC-OKLR/1.0/"))
-        case (Some("Undetermined"), _) => Some(URI("http://rightsstatements.org/vocab/UND/1.0/"))
-        case (Some("Unrestricted"), _) => Some(URI("http://rightsstatements.org/vocab/NoC-US/1.0/"))
+        case (Some("Restricted - Fully"), Some("Copyright")) => Some(rightsUri("InC"))
+        case (Some("Restricted - Fully"), Some("Donor Restrictions")) => Some(rightsUri("InC"))
+        case (Some("Restricted - Fully"), Some("Public Law 101-246")) => Some(rightsUri("InC"))
+        case (Some("Restricted - Fully"), Some("Service Mark")) => Some(rightsUri("NoC-OKLR"))
+        case (Some("Restricted - Fully"), Some("Trademark")) => Some(rightsUri("NoC-OKLR"))
+        case (Some("Restricted - Fully"), Some("Other")) => Some(rightsUri("NoC-OKLR"))
+        case (Some("Restricted - Partly"), Some("Copyright")) => Some(rightsUri("InC"))
+        case (Some("Restricted - Partly"), Some("Donor Restrictions")) => Some(rightsUri("InC"))
+        case (Some("Restricted - Partly"), Some("Public Law 101-246")) => Some(rightsUri("InC"))
+        case (Some("Restricted - Partly"), Some("Service Mark")) => Some(rightsUri("NoC-OKLR"))
+        case (Some("Restricted - Partly"), Some("Trademark")) => Some(rightsUri("NoC-OKLR"))
+        case (Some("Restricted - Partly"), Some("Other")) => Some(rightsUri("NoC-OKLR"))
+        case (Some("Restricted - Possibly"), Some("Copyright")) => Some(rightsUri("UND"))
+        case (Some("Restricted - Possibly"), Some("Donor Restrictions")) => Some(rightsUri("UND"))
+        case (Some("Restricted - Possibly"), Some("Public Law 101-246")) => Some(rightsUri("NKC"))
+        case (Some("Restricted - Possibly"), Some("Service Mark")) => Some(rightsUri("NoC-OKLR"))
+        case (Some("Restricted - Possibly"), Some("Trademark")) => Some(rightsUri("NoC-OKLR"))
+        case (Some("Restricted - Possibly"), Some("Other")) => Some(rightsUri("NoC-OKLR"))
+        case (Some("Restricted - Possibly"), _) => Some(rightsUri("UND"))
+        case (Some("Undetermined"), _) => Some(rightsUri("CNE"))
+        case (Some("Unrestricted"), _) => Some(rightsUri("NoC-US"))
         case (_, _) => None
       }}
 
     // The most restrictive statements
     val mostRestrictive = Seq(
```

Comment:
@moltude We might need a hierarchy rather than just a single "most restrictive". It occurs to me that you can have any combination of specific use restrictions with three different rights statements—UND, NKC, and NoC-OKLR. If you have UND plus either of the other two (or all three), UND is definitely the correct fallback. Don't we need to resolve the situation where you have NKC + NoC-OKLR (but not UND) as well?